homeassistant/components/wilight/switch.py

```diff
@@ -0,0 +1,352 @@
+        """Return trigger_3_description of valve switch."""
+        return wilight_to_hass_trigger(self._status.get("trigger_3"))
+
+    @property
+    def trigger_4_description(self) -> str | None:
+        """Return trigger_4_description of valve switch."""
+        return wilight_to_hass_trigger(self._status.get("trigger_4"))
+
+    @property
+    def state_attributes(self) -> dict[str, Any]:
+        """Return the state attributes."""
+        attr: dict[str, Any] = {}
+
+        if self.watering_time is not None:
+            attr[ATTR_WATERING_TIME] = self.watering_time
+
+        if self.trigger_1 is not None:
+            attr[ATTR_TRIGGER_1] = self.trigger_1
+
+        if self.trigger_2 is not None:
+            attr[ATTR_TRIGGER_2] = self.trigger_2
+
+        if self.trigger_3 is not None:
+            attr[ATTR_TRIGGER_3] = self.trigger_3
+
+        if self.trigger_4 is not None:
+            attr[ATTR_TRIGGER_4] = self.trigger_4
+
+        if self.trigger_1_description is not None:
+            attr[ATTR_TRIGGER_1_DESC] = self.trigger_1_description
+
+        if self.trigger_2_description is not None:
+            attr[ATTR_TRIGGER_2_DESC] = self.trigger_2_description
+
+        if self.trigger_3_description is not None:
+            attr[ATTR_TRIGGER_3_DESC] = self.trigger_3_description
+
+        if self.trigger_4_description is not None:
+            attr[ATTR_TRIGGER_4_DESC] = self.trigger_4_description
+
+        return attr
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use in the frontend."""
+        return ICON_WATERING
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the device on."""
+        await self._client.turn_on(self._index)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the device off."""
+        await self._client.turn_off(self._index)
+
+    async def async_set_switch_time(self, **kwargs: Any) -> None:
+        """Set the watering time."""
+        if ATTR_WATERING_TIME in kwargs:
+            target_time = kwargs[ATTR_WATERING_TIME]
+            await self._client.set_switch_time(self._index, target_time)
+
+    async def async_set_trigger(self, **kwargs: Any) -> None:
+        """Set the trigger according to index."""
+        if (ATTR_TRIGGER_INDEX in kwargs) & (ATTR_TRIGGER in kwargs):
+            trigger_index = kwargs[ATTR_TRIGGER_INDEX]
+            trigger = kwargs[ATTR_TRIGGER]
+            if trigger_index == 1:
+                await self._client.set_switch_trigger_1(self._index, trigger)
+            if trigger_index == 2:
+                await self._client.set_switch_trigger_2(self._index, trigger)
+            if trigger_index == 3:
+                await self._client.set_switch_trigger_3(self._index, trigger)
+            if trigger_index == 4:
+                await self._client.set_switch_trigger_4(self._index, trigger)
+
+
+class WiLightValvePauseSwitch(WiLightDevice, ToggleEntity):
```

Comment:
Please make the same changed for this entity as above comments.

Reply:
Done